man/man8/puppet.8

```diff
@@ -1,10 +1,19 @@
 .\" generated with Ronn/v0.7.3
-.\" https://github.com/rtomayko/ronn/tree/0.7.3
+.\" http://github.com/rtomayko/ronn/tree/0.7.3
 .
-.TH "PUPPET" "8" "May 2015" "Puppet Labs, LLC" "Puppet manual"
+.TH "PUPPET" "8" "January 2018" "Puppet Labs, LLC" "Puppet manual"
 .
 .SH "NAME"
 \fBpuppet\fR
 .
 .P
-[1;31mError: Could not parse application options: invalid option: \-\-help[0m
+Usage: puppet \fIsubcommand\fR [options] \fIaction\fR [options]
+.
+.P
+Available subcommands:
+.
+.P
+agent The puppet agent daemon apply Apply Puppet manifests locally ca Local Puppet Certificate Authority management\. (Deprecated) catalog Compile, save, view, and convert catalogs\. cert Manage certificates and requests certificate Provide access to the CA for certificate management\. certificate_request Manage certificate requests\. (Deprecated) certificate_revocation_list Manage the list of revoked certificates\. (Deprecated) config Interact with Puppet\'s settings\. describe Display help about resource types device Manage remote network devices doc Generate Puppet references epp Interact directly with the EPP template parser/renderer\. facts Retrieve and store facts\. file Retrieve and store files in a filebucket (Deprecated) filebucket Store and retrieve files in a filebucket generate Generates Puppet code from Ruby definitions\. help Display Puppet help\. inspect Send an inspection report key Create, save, and remove certificate keys\. (Deprecated) lookup Data in modules lookup function man Display Puppet manual pages\. master The puppet master daemon module Creates, installs and searches for modules on the Puppet Forge\. node View and manage node definitions\. parser Interact directly with the parser\. plugin Interact with the Puppet plugin system\. report Create, display, and submit reports\. resource The resource abstraction layer shell resource_type View classes, defined resource types, and nodes from all manifests\. status View puppet server status\. (Deprecated)
+.
```

Comment:
if you run `man -M ./man 8 puppet` does it show each application on a newline, or are all of the apps jammed together on one line?

Reply:
Hmmm they're all on one line, and the instances of `\.` all print as regular periods.

Reply:
Ah, so the old error message "Could not parse application options" was because `puppet --help` was broken, resulting in https://tickets.puppetlabs.com/browse/PUP-3251. I fixed `--help` awhile ago and we regenerated the man pages for 5.x in b36edbb613cf8ae72b9c4aef924bb68167ee6f45, but I don't think I touched 4.10.x.

That said, your PR matches current behavior elsewhere, so I'll merge this. Could you file a new ticket that the global applications are not displayed correctly and add it to epic PUP-7535?

Reply:
Could you also add another ticket to automatically generate man pages in CI like we do for localization, and add it to the epic?